internal/integrationtest/debug/debug_test.go

```diff
@@ -51,3 +51,46 @@ func TestDebuggerStarts(t *testing.T) {
 	_, _, err = cli.Run("debug", "-b", fqbn, "-P", programmer, sketchPath.String(), "--info")
 	require.NoError(t, err)
 }
+
+func TestDebuggerWithPdeSketchStarts(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("update")
+	require.NoError(t, err)
+
+	// Install core
+	_, _, err = cli.Run("core", "install", "arduino:samd")
+	require.NoError(t, err)
+
+	sketchName := "DebuggerPdeSketchStartTest"
+	sketchPath := cli.DataDir().Join(sketchName)
+	fqbn := "arduino:samd:mkr1000"
+
+	_, _, err = cli.Run("sketch", "new", sketchPath.String())
+	require.NoError(t, err)
+
+	// Looks for sketch file .ino
+	pathDir, err := sketchPath.ReadDir()
+	require.NoError(t, err)
+	fileIno := pathDir[0]
+
+	// Renames sketch file to pde
+	newFilePath := sketchPath.Join(sketchName + ".pde")
+	err = fileIno.Rename(newFilePath)
+	require.NoError(t, err)
+
+	// Looks for sketch file .pde
+	pathDir, err = sketchPath.ReadDir()
+	require.NoError(t, err)
+	filePde := pathDir[0]
```

Comment:
This is not required since you already have the patch of the `.pde` file inside `newFilePath`

Reply:
Right. In my mind I had to look again for the file since the one in the `fileIno` variable was always the same.
Thanks for pointing it out!